typed-transitions/src/Protocol/Core.hs

```diff
@@ -212,3 +212,34 @@ connect (PeerYield exchange n) (PeerAwait k) = case exchange of
 
 flipEither :: Either a b -> Either b a
 flipEither = either Right Left
+
+-- | Like 'connect' but allows to run holes even if one of the side completed.
+connect'
+  :: ( Monad m )
+  => Peer p tr (st begin)            endA m a
+  -> Peer p tr (Complement st begin) endB m b
+  -> m (Those a b)
```

Comment:
This is what `connect` should be. If you want, replace `connect` with this one.

Reply:
I will, it will be less confusing to just have one.